Governance doc (#1)
Tresjs/.github: GOVERNANCE.md

## Get Involved

**Anything that supports the TresJS community is a valuable contribution.**

Contributions might come in different forms, they are all meaningful and valuable to the community.

- Writing code (see our [contribution guidelines](CONTRIBUTING.md))
- Reporting a bug (with a proper reproduction link 😜)
- Suggesting an enhancement
- Documentation
- Translating docs to other languages
- Helping other users on Discord
- Creating tutorials or articles how you added 3D with TresJS to your Vue website

No contribution is too small, we are all here to learn and grow together.

You can contribute to TresJS no matter your engineering ability. Our goal is to recognize all contributors to the project regardless of skill, experience or background.

## Contributor Levels

### Level 1: Contributor 

Have you done any of the following? **Congratulations!, you are officially recognized as a contributor to the project!**

Examples of recognized contributions:

> - **Github** Filling a defailed bug report or feature request
> - **Github** Updating documentation or fixing a typo

Reviewer: ```suggestion
- **Github:** Filing a detailed bug report or feature request
- **Github:** Updating documentation or fixing a typo
```
I added colons to be uniform with sections further down in the document.

- Helping people on GitHub, Discord, etc.
- Answering questions on Stack Overflow, socials, etc.
- Creating tutorials, videos, courses and livestreams about TresJS

#### Accountabilities

Rest assured, this role does not require any extra responsibilities nor time commitment. We hope you stick around and keep participating in our community!

#### Nomination process

> You may self-nominate yourself bu sending the message `!contribute` to any Discord channel. If you do this, please share a second message with a link or a description of your contribution.

Reviewer 2: bu -> by

> You may also be granted this role automatically if you are acrive and helpful on Discord

Reviewer 2: acrive -> active

> ### Level 2: Maintainer

Reviewer 2: Maybe maintainer is a 2 level title? ##

Author: ## Contributor Levels 

is the level 2 title, all the levels inside need to be ###

The Maintainer role is available to contributors who want to join the team and take part in the long-term maintenance and growth of TresJS ecosystem.

This role is critical to TresJS long-term success and growth. Mantainers act as the public face of the project and are most likely to be your first point of contact when interacting with Discord or Github.

> [!NOTE]  
> **Maintainers are not required to write code**! Some Maintainers spend most of their time inside of Discord, maintaining a healthy community there. Others work on technical documentation, reviewing PRs, support, or design.

#### Recognized Contributions

There is no strict minimum number of contributions needed to reach this level, as long as you can show sustained involvement over some amount of time (at least a few months).

> - **GitHub**: Submitting multiple non-trivial pull requests and RFCs
> - **GitHub**: Reviewing multiple non-trivial pull requests and RFCs
> - **Discord**: Supporting users in Discord, especially in the #support channel
> - **Discord**: Active participation in RFC calls and other events
> - **GitHub + Discord**: Triaging and confirming user issues

Reviewer: ```suggestion
- **GitHub:**: Submitting multiple non-trivial pull requests and RFCs
- **GitHub:**: Reviewing multiple non-trivial pull requests and RFCs
- **Discord:**: Supporting users in Discord, especially in the #support channel
- **Discord:** Active participation in RFC calls and other events
- **GitHub + Discord:** Triaging and confirming user issues
```

This list is incomplete! Similar contributions are also recognized.

#### Privileges

- All privileges of the [Contributor role](#level-1-contributor), plus...
- Invitation to the `@mantainer` role in Discord

> - Invitation to the private `#maintainers` channel on Discord.
> - Invitation to the `TresJS` organization on GitHub.
> - Invitation to the `@maintainers` team on GitHub.

Reviewer: ```suggestion
- Invitation to the private `#maintainers` channel on Discord
- Invitation to the `TresJS` organization on GitHub
- Invitation to the `@maintainers` team on GitHub
```

- Ability to moderate Discord to remove spam, harmful speech, etc.
- Ability to push branches directly to the withTresJS GitHub organization (personal forks no longer needed). (Except for the `main` branch)

> - Ability to review GitHub PRs.
> - Ability to merge some GitHub PRs.
> - Ability to vote on some initiatives (see Voting below).

Reviewer: ```suggestion
- Ability to review GitHub PRs
- Ability to merge some GitHub PRs
- Ability to vote on some initiatives (see Voting below)
```

#### Responsabilities

- Access to `@tresjs` social media accounts.
- Access to organization Netlify account.
- Administrator privileges on the TresJs Github organization.
- Administration privileges on the TresJS Discord server.
- Publish access to all the organization's packages on NPM.
- Domain registrant and DNS access to the `tresjs.org` domain and subdomains.
- Ability to initiate a vote
- Ability to veto votes and resolve voting deadlocks
- Define project direction and planning
- Ability to decide on moderation decisions

#### Nomination

- Stewards cannot be self-nominated.
- Only Core members are eligible.
- New Stewards will be added based on a unanimous vote by the existing Steward(s).
- In the event that someone is unreachable then the decision will be deferred.

#### Retiring a Role (Alumni)

Contributor roles are granted for as long as the person wishes to engage with the project. However, over time an active community member may choose to step away from the TresJS ecosystem to work on other things. Moving on from a project is a natural and well-understood part of any open source community, and we celebrate it!

> Alumni is a special designation and role for any person who was once an active maintainer (L2 or above) but is now no longer actively involved. By retiring and joining Alumni you trade-in your current set of roles, privileges, and responsibilities for a new, special Alumni role (which comes with its own set of Privileges, as described above).

Reviewer: ```suggestion
Alumni is a special designation and role for any person who was once an active maintainer (L2 or above) but is now no longer actively involved. By retiring and joining Alumni you trade-in your current set of roles, privileges, and responsibilities for a new, special Alumni role (which comes with its own set of privileges, as described above).
```
